graphql/course/mutations.ts

```diff
@@ -106,6 +108,27 @@ export class MutateCourseResolver {
         }
         const result = await prisma.course.update({ data, where: { id: courseId } });
         logger.info(`Course (${result.id}) updated by Student (${context.user.studentId})`);
+
+        const subcourse = await prisma.subcourse.findFirst({ where: { courseId: courseId } });
+        const instructors = await prisma.subcourse_instructors_student.findMany({ where: { subcourseId: subcourse.id }, select: { student: true } });
+        const { context: contextData } = await prisma.user_achievement.findFirst({
+            where: { group: 'student_offer_course', context: { path: ['relation'], equals: `subcourse/${subcourse.id}` } },
```

Comment:
Do we need the group here? Maybe it's enough to filter by relation?

Reply:
This group selector is necessary. If we were not filtering for the specific group, every achievement that had subcourse/{{id}} as a relation would be added a courseName, even if it had no use for it.

Reply:
An other issue could be that the wrong contextData was fetched and data will be saved to the other achievement of an other group, missing essential information, previously existent.

Reply:
> This group selector is necessary. If we were not filtering for the specific group, every achievement that had subcourse/{{id}} as a relation would be added a courseName, even if it had no use for it.

The downside of referencing specific groups is that it might break completely over time if someone decides to rename the group. I guess having extra unnecessary data in the context doesn't hurt.

> An other issue could be that the wrong contextData was fetched and data will be saved to the other achievement of an other group, missing essential information, previously existent.

This could be solved by iterating over the achievements instead of updateMany

Reply:
Generally, we should try to not hardcode information that will be set dynamically in the database. The groups are used for grouping and not identification right now.

Reply:
Alright I fixed this part and removed the group from the filters.